tests/components/togrill/test_coordinator.py

```diff
@@ -0,0 +1,92 @@
+"""Test binary sensors for acaia integration."""
+
+from unittest.mock import Mock
+
+from bleak.exc import BleakError
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.togrill.coordinator import (
+    DeviceFailed,
+    DeviceNotFound,
+    ToGrillCoordinator,
+)
+from homeassistant.core import HomeAssistant
+
+from . import TOGRILL_SERVICE_INFO, setup_entry
+
+from tests.common import MockConfigEntry
+from tests.components.bluetooth import inject_bluetooth_service_info
+
+
+async def test_client_disconnect(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    mock_entry: MockConfigEntry,
+    mock_client: Mock,
+    mock_client_class: Mock,
+) -> None:
+    """Test that coordinator handles a disconnected client."""
+
+    inject_bluetooth_service_info(hass, TOGRILL_SERVICE_INFO)
+
+    await setup_entry(hass, mock_entry, [])
+
+    coordinator: ToGrillCoordinator = mock_entry.runtime_data
+    await coordinator.async_refresh()
+    assert coordinator.last_update_success
+
+    mock_client.is_connected = False
+    mock_client_class.connect.side_effect = BleakError("Failed to connect")
+    await coordinator.async_refresh()
+    assert not coordinator.last_update_success
+    assert isinstance(coordinator.last_exception, DeviceFailed)
```

Comment:
Generally we don't want to touch internals like entry.runtime_data. Tests generally test behaviour. I think this test should rather in a test that tests the result that the user sees. In this case, we can trigger raising the exception, and then we could see all the sensors to be unavailable

Reply:
Same goes for the other tests here, I personally believe coordinator tests are not the thing we should test, as the end user does not really care if we use a coordinator or that everything works per entity, as long as the entity behaves like expected

Reply:
I've dropped the file fully. It added coverage for 4 lines. So i dropped to 98% coverage. Which is fine by me.